### ISSUE TITLE
[Feature]: Add Logistic Regression/ Random Forest / XGBoost model toggle

### ISSUE THREAD

**Issue by VINAY-KUMAR855** [Arnav-Singh-5080/IPL-Win-Probability-Prediction#28]
Currently, the project uses a single ML model for predictions. Sometimes Logistic regression won't capture complex patters. so, i want to add Logistic Regression/ Random Forest / XGBoost model toggle Ui. So, User can switch multiple models . After selecting model, we train model, and show predictions.

**Comment by VINAY-KUMAR855** [comment]
I would like to work on this issue under GSSOC 2026. So, can you please assign to me.

**Comment by adyashaexe** [comment]
Hello @Arnav-Singh-5080, I would like to contribute to this issue as part of GSSOC 2026. I have gone through the issue details and would be happy to work on it. Kindly assign it to me.

**Comment by Manav20032008** [comment]
Hello @Arnav-Singh-5080  , i want to work on this project and my description is like this :-

# Title:-
Multi-Model Evaluation Toggle Using Machine Learning

### About Project
The multi-model evaluation toggle using machine learning algorithm will be to Prediction the most accurate model performance with the training scores of algorithms and the processing speed of classifiers with the any metric changes like precision-recall or any roc curves etc.., so basically machine learning will predict that total optimal outcome using algorithms and this will also aware all developers to not use less efficient models then expected.

The system analyzes algorithm performance, trains multiple ML models, and provides prediction results to help in early selection and architectural decision making.

### Technologies Used
Python
Machine Learning
Pandas
NumPy
Scikit-learn
XGBoost
Streamlit / Gradio (for Toggle UI)

### Features
Model performance and variance data analysis
Real-time accuracy toggle using ML algorithms
Confusion matrix and error evaluation of the model

### Algorithms Will be Used
Logistic Regression
Random Forest Classifier
XGBoost

### Conclusion
This project helps in predicting model efficacy at an early stage using Machine Learning techniques, improving feature selection and comparative analysis for developers.

**Comment by devanshyadav0001** [comment]
Hi maintainers
I would like to work on this feature under GSSoC 2026.

I have a good understanding of ML basics and can implement this project with proper data preprocessing, model training (Logistic Regression & Random Forest), and evaluation metrics. I will also ensure clean code and a well-documented README for beginners.

Please assign this issue to me. Looking forward to contributing! 🚀

**Comment by ujala100** [comment]
Hi @Arnav-Singh-5080

I’m interested in contributing to this project and would love to work on this issue. I have experience working on open-source contributions and would be happy to implement the required changes.

Could you please assign this issue to me? Thanks
